Mathlib/Combinatorics/SimpleGraph/Girth.lean

```diff
@@ -9,46 +9,55 @@ import Mathlib.Data.ENat.Lattice
 /-!
 # Girth of a simple graph
 
-This file defines the girth of a simple graph as the length of its smallest cycle, or `∞` if the
-graph is acyclic.
+This file defines the girth and the extended girth of a simple graph as the length of its smallest
+cycle, they give `0` or `∞` respectively if the graph is acyclic.
 -/
 
 namespace SimpleGraph
 variable {α : Type*} {G : SimpleGraph α} {n : ℕ∞}
 
-/-- The girth of a simple graph is the length of its smallest cycle, or `∞` if the graph is acyclic.
+/--
+The extended girth of a simple graph is the length of its smallest cycle, or `∞` if the graph is
+acyclic.
 -/
-noncomputable def girth (G : SimpleGraph α) : ℕ∞ :=
+noncomputable def egirth (G : SimpleGraph α) : ℕ∞ :=
 ⨅ a, ⨅ w : G.Walk a a, ⨅ _ : w.IsCycle, w.length
 
-@[simp] lemma le_girth : n ≤ G.girth ↔ ∀ a (w : G.Walk a a), w.IsCycle → n ≤ w.length := by
-  simp [girth]
+/--
+The girth of a simple graph is the length of its smallest cycle, or junk value `0` if the graph is
+acyclic.
+-/
+noncomputable def girth (G : SimpleGraph α) : ℕ :=
+ENat.toNat G.egirth
```

Comment:
```suggestion
noncomputable def girth (G : SimpleGraph α) : ℕ := G.egirth.toNat
```

Reply:
Unfortunately, this is resulting in the error:
```
invalid field notation, function 'ENat.toNat' does not have argument with type (ENat ...) that can be used, it must be explicit or implicit with a unique name
```

Reply:
Gladly! #13936
I'm not sure if the `#align`s of the old lemmas are okay this way or not.